gcp_variant_transforms/transforms/merge_conflicts.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Beam combiner function for merging VCF file header conflicts/definitions."""
+
+from collections import OrderedDict   #pylint: disable=unused-import
+
+import apache_beam as beam
+from gcp_variant_transforms.beam_io.vcf_header_io import VcfHeader
+from gcp_variant_transforms.libs.vcf_field_conflict_resolver import VcfParserConstants
+
+
+class _ConflictDetails(object):
+  """Container for conflict details for one field id."""
+  _MAX_FILE_NAME_NUM = 5
+
+  def __init__(self, definition, file_name):
+    # type: (OrderedDict[str, Union[str, int]], str) -> None
+    """Initialized a _ConflictDetails object.
+
+    Creates a dictionary that maps field definition to a list of file names.
+    Upon initialization, add ``definition`` and ``file_name`` to the dictionary.
+    """
+    self.conflicts = {self._format_definition(definition): list([file_name])}
+
+  def _add_details(self, other):
+    # type: (_ConflictDetails) -> None
+    for conflict, file_names in other.conflicts.iteritems():
+      if conflict not in self.conflicts.keys():
+        self.conflicts[conflict] = file_names
+        continue
+
+      for file_name in file_names:
+        if len(self.conflicts[conflict]) == self._MAX_FILE_NAME_NUM:
+          break
+        self.conflicts[conflict].append(file_name)
+
+  def _format_definition(self, definition):
+    # type: (OrderedDict[str, Union[str, int]]) -> str
+    """Formats the conflict string given one field definition.
+
+    Only num and type is considered for the conflicts, and the definition forms
+    a string in the format of `num={value} type={value}`.
+    """
+    conflict_keys = [VcfParserConstants.NUM, VcfParserConstants.TYPE]
+    formatted_conflict = []
+    for key in conflict_keys:
+      formatted_conflict.append(key + '=' + str(definition[key]))
+    return ' '.join(formatted_conflict)
+
+
+class VcfHeaderConflicts(object):
+  """Container for header conflicts."""
+
+  def __init__(self, vcf_header=VcfHeader()):
```

Comment:
Please do not set mutable objects as default arguments. See [here](http://docs.python-guide.org/en/latest/writing/gotchas/) for details. You can do `vcf_header = vcf_header or VcfHeader()` in the body of the method. However, I wonder if you should just check whether vcf_header is None and don't do anything in the `__init__`?